src/m365/aad/commands/app/app-add.ts

```diff
@@ -195,6 +213,94 @@ class AadAppAddCommand extends GraphCommand {
     return request.post<AppInfo>(createApplicationRequestOptions);
   }
 
+  private grantAdminConsent(appInfo: AppInfo, logger: Logger): Promise<AppInfo> {
+    return this.createServicePrincipal(appInfo.appId)
+      .then((sp: ServicePrincipalInfo) => {
+        if (this.debug) {
+          logger.logToStderr("Service principal created, returned object id: " + sp.id);
+        }
+
+        const tasks: Promise<void>[] = [];
+
+        // Parse resolved permissions
+        this.appPermissions.forEach(permission => {
+          if (permission.Scope.length > 0) {
+            // Send delegated permissions requests
+            tasks.push(this.grantOAuth2Permission(sp.id, permission.ResourceId, permission.Scope.join(' ')));
+            
+            if (this.debug) {
+              logger.logToStderr(`Admin consent for following resource ${permission.ResourceId}, with delegated permissions: ${permission.Scope.join(',')}`);
+            }
+          }
+
+          permission.resourceAccess.filter(access => access.type === "Role").forEach((access: ResourceAccess) => {
+            // Send application permissions requests
+            tasks.push(this.addRoleToServicePrincipal(sp.id, permission.ResourceId, access.id));
+
+            if (this.debug) {
+              logger.logToStderr(`Admin consent for following resource ${permission.ResourceId}, with application permission: ${access.id}`);
+            }
+          });
+        });
+    
+        return Promise.all(tasks)
+          .then(_ => {
+            return appInfo;
+          });
```

Comment:
we should add a `catch` with `handleRejectedODataJsonPromise`

Reply:
Is it really necessary? When testing the CLI by triggering explicitly the error, looks like it's already handled by the caller (`commandAction` method), just as the [client secret configuration](https://github.com/pnp/cli-microsoft365/pull/3455/files#diff-4bffc2f0d69d60fce428e83ed0b46c80a224eea3de200b3197330bdfb59a7fd0L542) behavior in case of failure.

Reply:
aaa yes. I agree that adding here handleReject would be an overkill as it will be done later on 👍.
thanks for checking this 👍. Will mark as resolved